### PR TITLE
Burger menu

### PR DESCRIPTION
**Pull request by awalsh3**
ISSUE: #6

This proposed PR adds a burger menu with drop-down links for mobile rendering. This uses tailwind css and plain vanilla javascript. 

**A few questions/notes:**

- It needs some styling decisions. I've left it as close to the main navbar as possible for now. Happy to adjust where needed.
- The drop-down links inherit the same classes used by the _links.html.erb shared partial. To change the styling we could either add another partial for mobile links or add code to the existing links partial if any adjustments are needed?
- I am having a little formatting issue with the drop-down vs the margin of the existing page. I am new to Turbo and not sure if there is something I am missing in the formatting. A quick fix is adding an id to the application template but I can work on a better solution. 
- Testing has all been manually done.  What is the protocol for testing? 

_A start for now! Please let me know what can be improved._

Photos below:

Screenshot 2023-08-03 at 17 57 03. 
  
Screenshot 2023-08-03 at 17 57 13. 

Screenshot 2023-08-03 at 17 57 25

### CODE REVIEW

File: app/javascript/controllers/navbar_controller.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+document.addEventListener('turbo:load', function () {
```
Comment (yshmarov) [inline app/javascript/controllers/navbar_controller.js:1]: at this point this is just vanilla js stored in a folder for stimulus controllers.

why don't you use an actual stimulus controller? :)

Comment (awalsh3) [reply app/javascript/controllers/navbar_controller.js:1]: Great point! I am new to stimulus but I _should_ follow the convention. I will work on it sometime this week. Thanks!

Comment (adrienpoly) [reply app/javascript/controllers/navbar_controller.js:1]: hello @awalsh3 thanks for taking this issue. 

I would like to suggest an alternative path at the same time with less JS or maybe none. the details/summary html tags can be used to create dropdowns. This is What Basecamp uses in Hey as an example 

Here is an article on that subject but they are also several others available :

https://dev.to/ayushn21/details-summary-tags-html-s-best-kept-secret-4ak4

Comment (adrienpoly) [reply app/javascript/controllers/navbar_controller.js:1]: When I mentioned Basecamp/Hey it was more for the technical part not really the way it looks. Your personal site looks clean I am sure you can make a good judgement on what looks best. I don't have a very strong option on this.

**Pull request closed without merging**